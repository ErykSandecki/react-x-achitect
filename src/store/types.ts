import { Store } from 'redux';

// types
import { TAction } from 'types/redux';
import { TMainState } from 'types/reducers';

export type TStore = Store<TMainState, TAction<any>> & {
  dispatch: unknown;
};
