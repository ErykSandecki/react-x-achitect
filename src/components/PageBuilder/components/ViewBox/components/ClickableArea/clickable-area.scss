@import '../../../../styles/z-indexes.scss';

.ClickableArea {
  height: 100%;
  overflow: visible;
  position: absolute;
  width: 100%;
  z-index: $--rx-zindex-events;

  &__outline {
    @include theme('blue-1', stroke);

    fill: rgba($color: #000000, $alpha: 0);
  }
}
