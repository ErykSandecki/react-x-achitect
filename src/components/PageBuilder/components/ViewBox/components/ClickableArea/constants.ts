export const SW = 2.5;
