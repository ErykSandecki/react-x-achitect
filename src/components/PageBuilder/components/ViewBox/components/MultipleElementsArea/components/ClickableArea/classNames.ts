export const className = 'ClickableArea';

export const classNames = {
  [className]: className,
  outline: `${className}__outline`,
};
