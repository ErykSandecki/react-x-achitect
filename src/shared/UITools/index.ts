// components
import ButtonIcon from './components/ButtonIcon/ButtonIcon';
import ColorPicker from './components/ColorPicker/ColorPicker';
import Section from './components/Section/Section';
import Tabs, { TTabsProps } from './components/Tabs/Tabs';

export const UITools = {
  ButtonIcon,
  ColorPicker,
  Section,
  Tabs,
};

export type TUITypes = {
  TTabsProps: TTabsProps;
};
