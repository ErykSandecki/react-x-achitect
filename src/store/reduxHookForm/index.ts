export * as actions from './actions';
export * as actionsType from './actionsType';
export * as reducer from './reducer';
export * as selectors from './selectors';
export * as types from './types';
