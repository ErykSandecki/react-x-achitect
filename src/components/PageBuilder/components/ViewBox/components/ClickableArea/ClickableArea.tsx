import { FC } from 'react';
import { useSelector } from 'react-redux';

// hooks
import { useClickableAreaEvents } from './hooks/useClickableAreaEvents';
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';
import { SW } from './constants';

// store
import { eventsSelector } from 'store/pageBuilder/selectors';

// styles
import styles from './clickable-area.scss';

// types
import { TRectCoordinates } from 'types';

export type TClickableAreaProps = {
  rectCoordinates: TRectCoordinates;
};

const ClickableArea: FC<TClickableAreaProps> = ({ rectCoordinates }) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const { isMultipleMoving } = useSelector(eventsSelector);
  const { x1, x2, y1, y2 } = rectCoordinates || {};
  const { ...events } = useClickableAreaEvents();

  if (isMultipleMoving) {
    return null;
  }

  return (
    <svg className={cx(classNamesWithTheme[className])}>
      <path
        className={cx(classNamesWithTheme.outline)}
        d={`M ${x1},${y1} H ${x2} V ${y2} H ${x1} V ${y1}`}
        strokeWidth={SW}
        {...events}
      />
    </svg>
  );
};

export default ClickableArea;
