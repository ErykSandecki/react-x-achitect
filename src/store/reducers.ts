import { combineReducers, Reducer } from 'redux';

// store
import appInitializer from './appInitializer/reducer';
import pageBuilder from './pageBuilder/reducer';
import reduxHookForm from './reduxHookForm/reducer';

// types
import { TMainState } from 'types/reducers';

const reducers = {
  appInitializer,
  pageBuilder,
  reduxHookForm,
};

const createRootReducer = (): Reducer<TMainState> => combineReducers(reducers);

export default createRootReducer;
