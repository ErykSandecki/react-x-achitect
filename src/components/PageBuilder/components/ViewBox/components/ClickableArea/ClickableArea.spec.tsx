import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import ClickableArea from './ClickableArea';

// mocks
import 'test/mocks/sagas/allSagas';

// others
import { BASE_RECT } from 'shared';

// store
import { configureStore } from 'store/store';

describe('ClickableArea snapshots', () => {
  it('should render ClickableArea', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <ClickableArea rectCoordinates={BASE_RECT} />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
