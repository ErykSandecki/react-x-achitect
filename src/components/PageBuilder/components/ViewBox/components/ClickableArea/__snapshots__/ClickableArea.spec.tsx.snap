// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickableArea snapshots should not render when isMultipleMoving 1`] = `<DocumentFragment />`;

exports[`ClickableArea snapshots should render ClickableArea 1`] = `
<DocumentFragment>
  <svg
    class="ClickableArea ClickableArea--light"
  >
    <path
      class="ClickableArea__outline ClickableArea__outline--light"
      d="M 0,0 H 0 V 0 H 0 V 0"
      stroke-width="1.5"
    />
  </svg>
</DocumentFragment>
`;
